sdk/rust/src/hbar.rs

```diff
@@ -0,0 +1,195 @@
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+
+use derive_more::{
+    Add,
+    AddAssign,
+    Div,
+    DivAssign,
+    Mul,
+    MulAssign,
+    Neg,
+    Sub,
+    SubAssign,
+};
+use rust_decimal::prelude::*;
+use serde_with::{
+    DeserializeFromStr,
+    SerializeDisplay,
+};
+
+use crate::Error;
+
+// TODO: add tests
+
+#[derive(SerializeDisplay, Copy, DeserializeFromStr, Hash, PartialEq, Eq, Clone, Add, Sub, Mul, Div, Neg, AddAssign, SubAssign, MulAssign, DivAssign)]
+pub struct Hbar(i64);
+
+pub type Tinybar = i64;
+
+#[repr(i64)]
+#[derive(Debug, SerializeDisplay, Copy, DeserializeFromStr, Hash, PartialEq, Eq, Clone)]
+pub enum HbarUnit {
+    /**
+     * The atomic (smallest) unit of hbar, used natively by the Hedera network.
+     * <p>
+     * It is equivalent to <sup>1</sup>&frasl;<sub>100,000,000</sub> hbar.
+     */
```

Comment:
Standardized formatting of doc comments would be something like
```suggestion
     /// The atomic (smallest) unit of [`Hbar`], used natively by the Hedera network.
     /// 
     /// It is equivalent to `1 / 100_000_000`  [`Hbar`].
```
The change on the last line is arguably style, but I'd note that the `code` type formatting makes it nice to understand.